src/cpp/src/visual_language/qwen2vl/classes.cpp

```diff
@@ -139,18 +194,44 @@ std::shared_ptr<ov::Model> patch_preprocess_into_model(std::shared_ptr<ov::Model
     reshape_shape2d->output(0).get_tensor().set_names({"reshape_shape2d"});
     auto image_mean = std::make_shared<ov::op::v0::Constant>(image_mean_tensor);
     auto image_scale = std::make_shared<ov::op::v0::Constant>(image_scale_tensor);
-    auto img_f32_nchw = create_f32_nchw_input(input_images);
-
-    auto img_resized = create_bicubic_resize(std::move(img_f32_nchw), resize_shape);
 
-    auto img_normalized = create_normalization(
-        std::move(img_resized), std::move(image_mean), std::move(image_scale)
-    );
+    // If
+    auto then_raw_image_1 = std::make_shared<ov::op::v0::Parameter>(ov::element::u8, ov::PartialShape{-1, -1, -1, -1});
+    auto then_resize_target_shape = std::make_shared<ov::op::v0::Parameter>(ov::element::i64, ov::PartialShape{2});
+    auto then_tile_shape = std::make_shared<ov::op::v0::Parameter>(ov::element::i64, ov::PartialShape{4});
+    auto model_then = patch_preprocess_branch_image(then_raw_image_1,
+                                                    then_resize_target_shape,
+                                                    image_mean,
+                                                    image_scale,
+                                                    then_tile_shape);
+
+    auto else_video = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, ov::Shape{1});
+    auto else_raw_frame_1 = std::make_shared<ov::op::v0::Parameter>(ov::element::u8, ov::PartialShape{-1, -1, -1, -1});
+    auto else_raw_frame_2 = std::make_shared<ov::op::v0::Parameter>(ov::element::u8, ov::PartialShape{-1, -1, -1, -1});
+    auto else_resize_target_shape = std::make_shared<ov::op::v0::Parameter>(ov::element::i64, ov::PartialShape{2});
+    auto model_else = patch_preprocess_branch_video(else_video,
+                                                    else_raw_frame_1,
+                                                    else_raw_frame_2,
+                                                    else_resize_target_shape,
+                                                    image_mean,
+                                                    image_scale);
+
+    auto if_op = std::make_shared<ov::op::v8::If>();
```

Comment:
Now you introduce `If` operation in the model graph. Will it change IR for already supported cases?
If yes, we can expect performance degradation for this supported cases on CPU because CPU is not happy about conditional ops.

@Wovchena, fyi.
@maxnick, could you please comment on `if` op if CPU plugin and GPU plugin support well this op in terms of performance